tests/integrations/ethereum/util/mock_rpc.go

```diff
@@ -109,15 +117,16 @@ func MockRpcClient(rpcMethods_ map[string]interface{}, callMethods_ map[string]m
 		response.Id = request.Id
 
 		// return expected response for that method, or failure if unimplemented
-		if request.Method != "eth_call" {
+		if method := request.Method; method != "eth_call" {
 			expectedResponse := rpcMethods[request.Method]
 
 			if expectedResponse != nil {
 				response.Result = expectedResponse
 				marshalResponse(response, rw)
 				return
 			} else {
-				marshalError(response, rw, "Unsupported method")
+				msg := fmt.Sprintf("Unsupported method, was %v, need eth_call", method)
```

Comment:
this comment is inaccurate - the method can be either `eth_call` or anything in `rpcMethods`